src/Playwright.Tests/BrowserContextRouteTests.cs

```diff
@@ -177,6 +177,65 @@ public async Task ShouldFallBackToContextRoute()
         Assert.AreEqual("context", await response.TextAsync());
     }
 
+
+    [PlaywrightTest("browsercontext-route.spec.ts", "should not throw on concurrent requests")]
+    public async Task ShouldNotThrowOnConcurrentRequest()
+    {
+        await using var context = await Browser.NewContextAsync();
+
+        // provide an html page which performs a high load of requests
+        await context.RouteAsync("**/empty.html", async (route) =>
+        {
+            await route.FulfillAsync(new()
+            {
+                Status = (int)HttpStatusCode.OK,
+                Body = @"<script>document.addEventListener('DOMContentLoaded', async () => { while (true) { await fetch('test.json') }})</script>"
+            });
+        });
+
+        // provide a simple json to fulfill the requests
+        await context.RouteAsync("**/test.json", async (route) =>
+        {
+            await route.FulfillAsync(new()
+            {
+                Status = (int)HttpStatusCode.OK,
+                Body = @"[]"
+            });
+        });
+
+
+        // subscribe for unobserved exceptions since we can not catch the exception
+        // in RouteHandler.cs directly
+        Exception? unobservedException = null;
+        void _handleUnobservedException(object sender, UnobservedTaskExceptionEventArgs args)
+            => unobservedException = args.Exception;
+
+        try
+        {
+            TaskScheduler.UnobservedTaskException += _handleUnobservedException;
```

Comment:
We have a similar logic already in place [here](https://github.com/microsoft/playwright-dotnet/blob/04431afbef20dfffe48022b8e953077526d14850/src/Playwright.Tests/Attributes/PlaywrightTestAttribute.cs#L139) so I'm wondering if we can remove 213-216 and 225-236. Keeping the Task.Delay seems essential but the rest is already covered? For me it fails constantly without the fix, so it seems tested?